src/queries/token/useGlobalStakingDataQuery.ts

```diff
@@ -0,0 +1,88 @@
+import { ethers } from 'ethers';
+import { useQuery, UseQueryOptions } from 'react-query';
+import { bigNumberFormatter } from 'thales-utils';
+import { GlobalStakingData } from 'types/token';
+import CCIPCollector from 'utils/contracts/ccipCollector';
+
+const useGlobalStakingDataQuery = (options?: UseQueryOptions<GlobalStakingData | undefined>) => {
+    return useQuery<GlobalStakingData | undefined>(
+        QUERY_KEYS.Token.GlobalStakingData(),
+        async () => {
+            const stakingData: GlobalStakingData = {
+                totalStakedAmount: 0,
+                thalesApy: 0,
+                feeApy: 0,
+                baseRewards: 0,
+                extraRewards: 0,
+            };
+            try {
+                // Thales staked - Base
+                const baseAnkrProvider = new ethers.providers.JsonRpcProvider(
+                    `https://rpc.ankr.com/base/${process.env.REACT_APP_ANKR_PROJECT_ID}`,
+                    Network.Base
+                );
+
+                const CCIPCollectorContract = new ethers.Contract(
+                    CCIPCollector.addresses[Network.Base],
+                    CCIPCollector.abi,
+                    baseAnkrProvider
+                );
+
+                const period = Number(await CCIPCollectorContract.period()) - 1;
+                const [
+                    price,
+                    stakedAmount,
+                    escrowedAmount,
+                    calculatedRevenueForPeriod,
+                    baseRewardsPool,
+                    bonusRewardsPool,
+                ] = await Promise.all([
+                    fetch(`${generalConfig.API_URL}/token/price`),
+                    CCIPCollectorContract.calculatedStakedAmountForPeriod(Number(period)),
+                    CCIPCollectorContract.calculatedEscrowedAmountForPeriod(Number(period)),
+                    CCIPCollectorContract.calculatedRevenueForPeriod(Number(period)),
+                    CCIPCollectorContract.baseRewardsPerPeriod(),
+                    CCIPCollectorContract.extraRewardsPerPeriod(),
+                ]);
+
+                const revShare = period <= 3 ? 30000 : bigNumberFormatter(calculatedRevenueForPeriod);
+                const thalesTokenPrice = Number(await price.text());
+
+                stakingData.feeApy =
+                    (revShare * 52 * 100) /
+                    ((bigNumberFormatter(stakedAmount) + bigNumberFormatter(escrowedAmount)) * thalesTokenPrice);
+
+                const thalesRewardsAPY =
+                    ((bigNumberFormatter(baseRewardsPool) + bigNumberFormatter(bonusRewardsPool)) * 52 * 100) /
+                    (bigNumberFormatter(stakedAmount) + bigNumberFormatter(escrowedAmount));
+
+                stakingData.feeApy = aprToApy(stakingData.feeApy);
+                stakingData.thalesApy = aprToApy(thalesRewardsAPY);
+
+                stakingData.feeApy = Number((Math.round(stakingData.feeApy * 100) / 100).toFixed(2));
```

Comment:
Why * 100 and / 100?

Reply:
Stack overflow answer to round up to 2 decimals with all edge cases covered.